### PR TITLE
Release 2.3.1.2

### PR DESCRIPTION
**Pull request by bskinn**

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -8,6 +8,56 @@ and this project follows an extension of
...
+    template was missing parentheses to perform a method call; this
+    caused the template rendering to emit a Python string describing
+    the parent template object, instead of rendering the parent
+    template as intended.
+    ([#298](https://github.com/bskinn/sphobjinv/issues/298))
+
+#### Internal
+
+  * Moved the Sphinx linkcheck job out of CI and into `tox`.
+    * The linkcheck is often flaky, and is a nuisance when it fails the CI.
```
Comment (jayaddison) [inline CHANGELOG.md:25]: (reviewing the changelog out of interest)

Is it usually/always the `opensource.org` MIT license hyperlink that fails here?  As found at: https://dev.azure.com/brianskinn/sphobjinv/_build/results?buildId=539&view=logs&j=ce6756cd-8cff-5301-9ad1-3614e3437a06&t=ce14a895-24aa-5627-5752-833db68b916f#L137

(I don't think the line-number fragment link works there.. but adding it for reference anyway)

I'm slightly surprised by that HTTP 403 response code; if it happens frequently enough to be disruptive, though -- and if there aren't other problems -- then appending it to `linkcheck_ignore` in `conf.py` could be an option.

Or, as you say, release-time is a reasonable cadence for checking the hyperlinks, anyway.

Comment (bskinn) [reply CHANGELOG.md:25]: No, it varies. Most often, it's been a random Sphinx docset. This is a new problem that emerged a couple of months ago; I'm pretty sure that one or more CDNs/web frameworks/WSGI tools/etc. made a change to try to cut down on spam requests, and they've started treating requests out of CI systems much more suspiciously.

The linkcheck often runs fine from my local machine, but consistently fails with 403 Forbidden in CI.

Comment (bskinn) [reply CHANGELOG.md:25]: The original place I noticed this was in my repo with a workflow to validity-check a Gist with `intersphinx_mapping` entries.

Previously, it just used `sphobjinv`'s `--url` mode to ingest the respective web-hosted `objects.inv` files directly. But, when that started returning HTTP 403 for some requests, I [switched][pr] to downloading with `requests` and then ingesting from local, which made things happy enough.

[pr]: https://github.com/bskinn/intersphinx-gist/pull/15/files

**Comment by bskinn** [comment]
@jayaddison 2.3.1.2 is up on PyPI - hopefully this fixes the reproducibility issue, please open another ticket if not.

**Pull request merged**